### PR TITLE
Stop using platform-specific names for props

### PR DESCRIPTION
**Pull request by odino**
Follows up on [this comment](https://github.com/facebook/react-native/pull/5065#issuecomment-168353782) by @nicklockwood in #5065.

Rely on using the @platform annotation and keep the API / naming less redundant.

### CODE REVIEW

**Comment by nicklockwood** [comment]
I agree (of course) on the basic premise.

Instead of a breaking change, it might be better to add some internal JS logic to detect if javascriptEnabledAndroid is being set and map it to javascriptEnabled, with a console.warn explaining the old prop is deprecated.

File: breaking-changes.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## 0.17
 
+- `javaScriptEnabledAndroid` of the `WebView` component has been renamed to `javaScriptEnabled`
```
Comment (astreet) [inline breaking-changes.md]: Add that domStorageEnabledAndroid was also renamed

**Comment by astreet** [comment]
@facebook-github-bot shipit

File: breaking-changes.md (hunk 1 above)
Comment (odino) [inline breaking-changes.md]: Won't it create confusion? No one is really using that prop as its been
added a few commits back -- though I agree it wont hurt :)

**Comment by nicklockwood** [comment]
@odino don't worry, I've added some internal warnings and remapped the old props to the new ones as I suggested, so there are now no breaking changes to contend with.

**Comment by odino** [comment]
@nicklockwood thanks man!

**Pull request closed without merging**